### PR TITLE
Feature/128 introduce todo component

### PR DESCRIPTION
**Pull request by Nicolai-Bak**

### CODE REVIEW

**Review by AntonEsbjergIT-Minds — APPROVED** [review]
Looks great, functionality works and code is clean

File: todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  templateUrl: './todo.component.html',
+  styleUrls: ['./todo.component.scss']
+})
+export class TodoComponent implements OnInit {
+  deadline: string = '';
+  constructor() { }
+
+  @Input() todo?: Todo;
+
+  @Output() onCheckboxClick = new EventEmitter<boolean>();
```
Comment (PRP-ITM) [inline todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts]: This chould be named: checkboxClick then the eventhandler should be named onCheckboxClick :)

File: todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  ngOnInit(): void {
+    this.deadline = this.todo?.deadline?.split('T')[0] ?? "";
```
Comment (PRP-ITM) [inline todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts:18]: Is this a date? :D

File: todo-frontend-angular/src/app/todolist-selector/todolist/todolist.component.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<section>
+  <div class="btns-container">
+    <app-button [buttonConfig]="textBtnConfig" (onTextBtnClick)="onClickEventReceived()">
+    </app-button>
+  </div>
+</section>
+
+<div class="todos">
+  <div *ngFor="let todo of todos">
```
Comment (PRP-ITM) [inline todo-frontend-angular/src/app/todolist-selector/todolist/todolist.component.html]: If you don't use the div for styling, you could just add the *ngFor to the <app-todo> element :)

File: todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts (hunk 2 above)
Comment (Nicolai-Bak) [reply todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts:18]: yes.. replaced with pipe in later pull request

File: todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts (hunk 1 above)
Comment (Nicolai-Bak) [reply todo-frontend-angular/src/app/todolist-selector/todo/todo.component.ts]: updated

File: todo-frontend-angular/src/app/todolist-selector/todolist/todolist.component.html (hunk 3 above)
Comment (Nicolai-Bak) [reply todo-frontend-angular/src/app/todolist-selector/todolist/todolist.component.html]: redundant div removed

**Review by PRP-ITM — APPROVED** [review]

**Pull request merged**